### DISCUSSION TITLE
Implementing custom processor method without forking

### DISCUSSION THREAD

**Discussion by Luko6** [SixLabors/ImageSharp#1514] [Q&A] [answered]
Hi!

I would like to implement my own parallelization in the ResizeProcessor, replacing the ParallelRowIterator.
Is it possible to do so withot forking the ImageSharp repo, or am i stuck with that?

**Comment by antonfirsov** [comment, marked as answer]
We don't have extension points for processors, but instead of forking I would strongly recommend to copy the code of `ResizeProcessor` and `ResizeProcessor<T>` and use `imageProcessingContext.ApplyProcessorn(new MyResizeProcessor(...))`.

Out of curiosity: why do you want to replace `ParallelRowIterator` ? Is there anything we can improve there instead?

**Reply by Luko6** [reply]
I am currently working with Hastlayer on [this](https://github.com/SixLabors/ImageSharp/discussions/1352), and `Parallel.For()` is not supported in Hastlayer
